pytorch_lightning/trainer/data_loading.py

```diff
@@ -212,6 +241,19 @@ def __init__(self, num_features, dataset, *args, **kwargs):
 
         if not contains_dataset:
             dl_args.pop('dataset')
+        else:
```

Comment:
Should this be `if contains_iterable_dataset`? And remove the modification to the `if` above